### PR TITLE
feat: ignore capitalization for authentication

### PR DESCRIPTION
**Pull request by m0nhawk**
Link to JIRA ticket if there is one: [VADC-1233](https://ctds-planx.atlassian.net/browse/VADC-1233)

### New Features
* case insensitive comparison for usernames

### Deployment changes

[VADC-1233]: https://ctds-planx.atlassian.net/browse/VADC-1233?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jacob50231 — APPROVED** [review]
SQL changes all look good. Go changes seem minimal. All checks out!

File: arborist/auth.go [hunk 1]
```diff
@@ -678,12 +678,12 @@ func authMappingForUser(db *sqlx.DB, username string) (AuthMapping, *ErrorRespon
...
 			SELECT usr_policy.policy_id FROM usr
 			INNER JOIN usr_policy ON usr_policy.usr_id = usr.id
-			WHERE usr.name = $1 AND (usr_policy.expires_at IS NULL OR NOW() < usr_policy.expires_at)
+			WHERE LOWER(usr.name) = $1 AND (usr_policy.expires_at IS NULL OR NOW() < usr_policy.expires_at)
 			UNION
 			SELECT grp_policy.policy_id FROM usr
 			INNER JOIN usr_grp ON usr_grp.usr_id = usr.id
 			INNER JOIN grp_policy ON grp_policy.grp_id = usr_grp.grp_id
-			WHERE usr.name = $1 AND (usr_grp.expires_at IS NULL OR NOW() < usr_grp.expires_at)
+			WHERE LOWER(usr.name) = $1 AND (usr_grp.expires_at IS NULL OR NOW() < usr_grp.expires_at)
```
Comment (paulineribeyre) [inline arborist/auth.go:686]: this function is called by `handleAuthMappingPOST` and `handleAuthMappingGET`, you lowered the username in `handleAuthMappingGET` only. If i'm reading this right, either the `username` on L701 should be lowered too, or `username` should be lowered in `handleAuthMappingPOST`

File: arborist/user.go [hunk 2]
```diff
@@ -164,7 +164,7 @@ func fetchUserPolicyInfo(db *sqlx.DB, user_name string, policy_name string) (*Us
 			FROM usr_policy
 			INNER JOIN policy ON policy.id = usr_policy.policy_id
 			INNER JOIN usr on usr_policy.usr_id = usr.id 
-			WHERE usr.name like $1 and policy.name like $2;
+			WHERE LOWER(usr.name) like $1 and policy.name like $2;
```
Comment (paulineribeyre) [inline arborist/user.go:167]: should `user_name` on L173 be lowered?

**Review by paulineribeyre — CHANGES_REQUESTED** [review]

**Review by paulineribeyre — APPROVED** [review]
this looks good, but i didn't go over all the existing functions to make sure we didn't miss usernames that should be in lowercase - ideally we should do that

File: arborist/user.go (hunk 2 above)
Comment (m0nhawk) [reply arborist/user.go:168]: Addressed in this commit: https://github.com/uc-cdis/arborist/pull/165/commits/792db81455b841cea4226fa20e00a214278b26ce

File: arborist/auth.go (hunk 1 above)
Comment (m0nhawk) [reply arborist/auth.go:686]: Addressed in this commit: https://github.com/uc-cdis/arborist/pull/165/commits/0019600338c3c26a579b66bd8333226975f542f2

**Pull request merged**